common/src/stack/command/stack/argument_processors/pallet.py

```diff
@@ -1,6 +1,17 @@
-from collections import namedtuple
+from dataclasses import dataclass
 from stack.exception import ArgNotFound
 
+@dataclass(eq=True, order=True, frozen=True)
+class Pallet:
+	"""A dataclass representation of the Roll schema in the database."""
+	id: int
+	name: str
+	version: str
+	rel: str
+	arch: str
+	os: str
+	url: str
+
```

Comment:
For my own education: What use a dataclass instead of a namedtuple? Besides increasing your contribution line count before leaving :p

Reply:
It's like you didn't read the description at all... This is split out from changes being made in `feature/hooks` since it seemed like a small reviewable chunk in its own right. I ended up using a dataclass over there thus it is a dataclass over here.

https://github.com/Teradata/stacki/blob/a6d9d48633cf4e3658cf5a519d3c0343c5c8a2f0/common/src/stack/command/stack/argument_processors/pallet.py#L40

Reply:
Thar' be a lot of black magic in that `create_from_probepal_pallet_info` method. 
```
		return cls(
			id=0,
			**{
				key: getattr(pallet_info, PALLET_ATTR_MAPPINGS[key])
				for key in PALLET_ATTR_MAPPINGS
			},
			url="",
		)
```
Makes me think back to my Perl days